reload/reload_test.go

```diff
@@ -0,0 +1,308 @@
+
+				Expect(nonReloadableProcess.Direct).To(Equal(false))
+				Expect(nonReloadableProcess.Default).To(Equal(false))
+
+				Expect(reloadableProcess.Direct).To(Equal(false))
+				Expect(reloadableProcess.Default).To(Equal(true))
+			})
+		})
+
+		context("when direct is true and default is false", func() {
+			it.Before(func() {
+				originalProcess.Direct = true
+				originalProcess.Default = false
+			})
+
+			it("should return true, false, true, false", func() {
+				nonReloadableProcess, reloadableProcess := reload.TransformReloadableProcesses(originalProcess, watchExecProcessSpec)
+
+				Expect(nonReloadableProcess.Direct).To(Equal(true))
+				Expect(nonReloadableProcess.Default).To(Equal(false))
+
+				Expect(reloadableProcess.Direct).To(Equal(true))
+				Expect(reloadableProcess.Default).To(Equal(false))
+			})
+		})
+
+		context("when direct is true and default is true", func() {
+			it.Before(func() {
+				originalProcess.Direct = true
+				originalProcess.Default = true
+			})
+
+			it("should return true, false, true, true", func() {
+				nonReloadableProcess, reloadableProcess := reload.TransformReloadableProcesses(originalProcess, watchExecProcessSpec)
+
+				Expect(nonReloadableProcess.Direct).To(Equal(true))
+				Expect(nonReloadableProcess.Default).To(Equal(false))
+
+				Expect(reloadableProcess.Direct).To(Equal(true))
+				Expect(reloadableProcess.Default).To(Equal(true))
+			})
+		})
+
+		context("when spec.Verbose is 1", func() {
+			it.Before(func() {
+				watchExecProcessSpec.Verbose = 1
+			})
+
+			it("should return args with 1 verbose flag", func() {
+				expectedReloadableProcess.Args = []string{
+					"--restart",
+					"--watch", "watch-dir0",
+					"--watch", "watch-dir1",
+					"--shell", "my-shell",
+					"-v",
+					"--",
+					"my-command",
+					"original-arg0",
+					"original-arg1",
+				}
+
+				nonReloadableProcess, reloadableProcess := reload.TransformReloadableProcesses(originalProcess, watchExecProcessSpec)
+
+				Expect(nonReloadableProcess).To(Equal(expectecNonReloadableProcess))
+				Expect(reloadableProcess).To(Equal(expectedReloadableProcess))
+			})
+		})
+
+		context("when spec.Verbose is 4", func() {
+			it.Before(func() {
+				watchExecProcessSpec.Verbose = 4
+			})
+
+			it("should return args with 4 verbose flag", func() {
+				expectedReloadableProcess.Args = []string{
+					"--restart",
+					"--watch", "watch-dir0",
+					"--watch", "watch-dir1",
+					"--shell", "my-shell",
+					"-vvvv",
+					"--",
+					"my-command",
+					"original-arg0",
+					"original-arg1",
+				}
+
+				nonReloadableProcess, reloadableProcess := reload.TransformReloadableProcesses(originalProcess, watchExecProcessSpec)
+
+				Expect(nonReloadableProcess).To(Equal(expectecNonReloadableProcess))
+				Expect(reloadableProcess).To(Equal(expectedReloadableProcess))
+			})
+		})
+	})
+
+	context("WatchExecRequirement", func() {
+		it("should require watchexec", func() {
```

Comment:
nit: i think this test would be clearer if it read: "it should require watchexec at launch"

Reply:
you are correct.